uk.co.compucorp.civicrm.hrcore/gulp/tasks/requirejs.js

```diff
@@ -0,0 +1,182 @@
+var colors = require('ansi-colors');
+var detectInstalled = require('detect-installed');
+var exec = require('child_process').exec;
+var find = require('find');
+var fs = require('fs');
+var gulp = require('gulp');
+var gulpSequence = require('gulp-sequence');
+var path = require('path');
+
+var test = require('../test');
+var utils = require('../utils');
+
+var originalExtension;
+
+module.exports = [
+  {
+    name: 'requirejs',
+    fn: requireJsTask
+  },
+  {
+    name: 'requirejs:watch',
+    fn: function (cb) {
+      var extPath, watchPatterns;
+
+      if (utils.canCurrentExtensionRun('requirejs')) {
+        extPath = utils.getExtensionPath();
+        watchPatterns = utils.addExtensionCustomWatchPatternsToDefaultList([
+          path.join(extPath, '**', 'src/**/*.js')
+        ], 'requirejs');
+
+        gulp.watch(watchPatterns, ['requirejs']).on('change', function (file) {
+          if (utils.canCurrentExtensionRun('test')) {
+            try {
+              test.for(file.path);
+            } catch (ex) {
+              test.all();
+            }
+          }
+        });
+        cb();
+      } else {
+        console.log(colors.yellow('Not eligible for this task, skipping...'));
+        cb();
+      }
+    }
+  }
+];
+
+/**
+ * It scans the build.js file of all the CiviHR extension, to check if in any of them
+ * the current extension is marked as a dependency.
+ *
+ * If any are found, then the `requirejs` task is executed for the extension
+ * the build.js file belongs to.
+ *
+ * The task is recursive, stopping when no further dependencies are found
+ *
+ * @param {Function} cb
+ */
+function extensionDependenciesTask (cb) {
+  var buildFiles = find.fileSync(/js(\/[^/]+)?\/build\.js$/, path.join(__dirname, '../../../'));
+
+  var sequence = buildFiles.filter(function (buildFile) {
+    var content = fs.readFileSync(buildFile, 'utf8');
+
+    return (new RegExp(utils.getCurrentExtension(), 'g')).test(content);
```

Comment:
'g' is not needed. did you mean 'i' by any chance?

Reply:
no, 'g' for matching the regex on multiple lines. without it, it doesn't work